src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginDisplay.java

```diff
@@ -1,61 +1,83 @@
 package com.github.manolo8.darkbot.gui.plugins;
 
 import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.extensions.plugins.Plugin;
 import com.github.manolo8.darkbot.extensions.plugins.PluginHandler;
 import com.github.manolo8.darkbot.extensions.plugins.PluginListener;
+import com.github.manolo8.darkbot.extensions.plugins.PluginUpdater;
 import com.github.manolo8.darkbot.gui.components.MainButton;
 import com.github.manolo8.darkbot.gui.utils.UIUtils;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
-import java.awt.*;
+import java.util.Arrays;
 import java.util.stream.Stream;
 
 public class PluginDisplay extends JPanel implements PluginListener {
 
     private Main main;
     private MainButton pluginTab;
     private PluginHandler pluginHandler;
+    private PluginUpdater pluginUpdater;
 
     private JPanel pluginPanel;
 
     public PluginDisplay() {
-        super(new BorderLayout());
+        super(new MigLayout("wrap 1, fillx", "[fill]", ""));
         setBorder(BorderFactory.createEmptyBorder());
     }
 
     public void setup(Main main, MainButton pluginTab) {
         this.main = main;
         this.pluginTab = pluginTab;
         this.pluginHandler = main.pluginHandler;
-        if (pluginPanel == null) add(setupUI());
+        this.pluginUpdater = main.pluginUpdater;
+        pluginUpdater.setPluginDisplay(this);
+        setupUI();
         refreshUI();
         pluginHandler.addListener(this);
     }
 
-    private JComponent setupUI() {
+    private void setupUI() {
+        if (pluginPanel != null) return;
         pluginPanel = new JPanel(new MigLayout("wrap 1, fillx", "[fill]", ""));
-        JScrollPane scrollPane = new JScrollPane(pluginPanel, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
 
-        scrollPane.setBorder(BorderFactory.createEmptyBorder());
-        scrollPane.getVerticalScrollBar().setUnitIncrement(25);
-        return scrollPane;
+        add(new PluginUpdateHeader(pluginUpdater));
+        add(new ScrollPane(pluginPanel));
```

Comment:
There is absolutely no need to subclass JScrollPane here, it's a pretty concise 3-line code used only on one place
```suggestion
        JScrollPane scrollPane = new JScrollPane(pluginPanel, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);

        scrollPane.setBorder(BorderFactory.createEmptyBorder());
        scrollPane.getVerticalScrollBar().setUnitIncrement(25);
        add(scrollPane);
```